docs/jotai/api/devtools.mdx

```diff
@@ -44,3 +44,84 @@ export const useTasksDevtools = () => {
   useAtomDevtools(tasksAtom, 'Tasks (Instance 2)')
 }
 ```
+
+## useAtomsSnapshot
+
+⚠️ Note: This hook only works in a `process.env.NODE_ENV !== 'production'` environment.
+
+`useAtomsSnapshot` takes a snapshot of the currently mounted atoms and their state.
+
+```ts
+function useAtomsSnapshot(scope?: Scope): AtomsSnapshot
+```
+
+It accepts an atom `scope` parameter and will return an `AtomsSnapshot`, which is basically a `Map<AnyAtom, unknown>`. You can use the `Map` API to iterate over atoms and their state.
+This hook is primarily meant for debugging and devtools use cases.
+
+Be careful using this hook because it will cause the component to re-render for all state changes.
+
+### Example
+
+```ts
+import { Provider } from 'jotai'
+import { useAtomsSnapshot } from 'jotai/devtools'
+
+const RegisteredAtoms: React.FC = () => {
+  const atoms = useAtomsSnapshot()
```

Comment:
@Thisen This hook shouldn't be used outside Provider in L73, or it will be using a different store.